handler/src/main/java/io/netty/handler/ssl/OpenSslServerSessionContext.java

```diff
@@ -32,36 +34,73 @@ public void setSessionTimeout(int seconds) {
         if (seconds < 0) {
             throw new IllegalArgumentException();
         }
-        SSLContext.setSessionCacheTimeout(context.ctx, seconds);
+        Lock writerLock = context.ctxLock.writeLock();
+        try {
+            writerLock.lock();
```

Comment:
move the lock statement outside the `try`. If `lock` fails we shouldn't try to `unlock`.

This is a general comment when the locks are used.